Lesson_12/Tusk_1.py

```diff
@@ -0,0 +1,87 @@
+import json
+
+
+print('Phone list program is started.')
+
+commands = ('You can use next commands:'
+            '\n  - stats: If you want to see the quantity of records'
+            '\n  - add: if you want to add a new record'
+            '\n  - delete <name>: if you want to delete a record with name in <>'
+            '\n  - list: to see the list of all records'
+            '\n  - show <name>: to see the information about the name in <>'
+            '\n  - commands: if you want to see tha command list again'
+            '\n  - exit: if you want to exit the program')
+print(commands)
+file_address = 'phone_book.txt'
+
+while True:
+    with open(file_address, 'r') as file:
+        json_data = file.read()
+    if json_data:
+        phone_list = json.loads(json_data)
+    else:
+        phone_list = {}
+
+    command = input('Please enter a command: ').lower()
+
+    if command == 'stats':
+        if len(phone_list) == 0:
+            print('There is no records for now')
+        else:
+            print(len(phone_list))
+
+    elif command == 'add':
+        name = input('Please enter a name: ')
+        if name in phone_list:
+            print('This name is already here')
+            continue
+        phone_nr = input('Please enter a number: ')
+        if phone_nr.isdigit():
+            phone_list[name] = phone_nr
+            print(f'The name {name} added')
```

Comment:
Your program can be closed by keyboard interruption and I think you wand to keep all user-entered data. That's why I would suggest writing to a file in places like this, when you are modifying your data. There are only two such places: when you add something and delete something